Task10_Auth/src/test/java/ZipCodesTests.java

```diff
@@ -57,12 +63,43 @@ public void addDuplicatesZipCodesAndCheckThereAreNoDuplicationsTest() {
         expectedZipCodesAfterPost.addAll(newZipCodes);
 
         ResponseWrapper responsePost = zipCodesClient.postZipCodes(newZipCodes);
-        List<String> resultZipCodes = responsePost.getResponseBody();
+        List<String> resultZipCodes = new ArrayList<>(responsePost.getResponseBody());
         List<String> resultZipCodesWithoutDuplications = new ArrayList<>(new HashSet<>(resultZipCodes));
 
         assertAll("Status code is 201 // Zip codes from request body are added to available zip codes //  There are no duplications in available zip codes",
                 () -> assertEquals(expectedZipCodesAfterPost, resultZipCodes),
                 () -> assertEquals(resultZipCodesWithoutDuplications, resultZipCodes),
                 () -> assertEquals(201, responsePost.getResponseCode()));
     }
+
+    @Test
+    public void addAlreadyUsedZipCodeAndCheckThereIsNoDuplicationInUsedAndNotUsedZipCodes() {
+        UsersClient usersClient = new UsersClient();
+        UsersClient.createUserIfNotExist(usersClient);
+        ResponseWrapper responseGet = zipCodesClient.getZipCodes();
+        List<String> availableZipCodes = new ArrayList<>(responseGet.getResponseBody());
+        ResponseWrapper getUsersResponse = usersClient.getUsers();
+        List<UsersDto> usersList = new ArrayList<>(getUsersResponse.getResponseBody());
+        Random random = new Random();
+        List<String> newZipCodes = new ArrayList<>();
+        newZipCodes.add(usersList.get(random.nextInt(usersList.size())).getZipCode());
+        newZipCodes.add(usersList.get(random.nextInt(usersList.size())).getZipCode());
```

Comment:
These zip codes can be equal to null and in this case test will be incorrect

Reply:
not relevant